### DISCUSSION TITLE
increase timeout

### DISCUSSION THREAD

**Discussion by alienatorZ** [micro/go-micro#2598] [Q&A] [answered]
Hello!
I am trying to push a file to a service to process and it takes awhile so I am getting Context Deadline Exceeded from the calling clients. Is there a way to change the timeout when calling a service?

Thanks

**Comment by alienatorZ** [comment, marked as answer]
I think I solved this with `err := m.svc.Client().Call(context.TODO(), req, rsp, client.WithRequestTimeout(90*time.Second))`

**Reply by Davincible** [reply]
Yes, that works. Or you can set the default timeout;

https://github.com/go-micro/go-micro/blob/0a91ba7b5d7319ae71e551c21fe5134604490835/client/options.go#L23
